azure/services/agentpools/spec.go

```diff
@@ -278,6 +281,7 @@ func (s *AgentPoolSpec) Parameters(existing interface{}) (params interface{}, er
 			EnableAutoScaling:    s.EnableAutoScaling,
 			EnableUltraSSD:       s.EnableUltraSSD,
 			KubeletConfig:        kubeletConfig,
+			KubeletDiskType:      containerservice.KubeletDiskType(to.String(s.KubeletDiskType)),
```

Comment:
Since it looks like AKS doesn't allow this to be changed, we should add a check in the webhook like:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/8b6f01a615a639df20644ffd0928d778b5e9fc9e/exp/api/v1beta1/azuremanagedmachinepool_webhook.go#L120-L125